examples/tagmap/tagmap-layer/text-layer/multi-icon-layer/multi-icon-layer-vertex-64.glsl.js

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2015 - 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+export default `\
+#define SHADER_NAME multi-icon-layer-vertex-shader-64
+
+attribute vec2 positions;
+
+attribute vec3 instancePositions;
+attribute vec2 instancePositions64xyLow;
+attribute float instanceSizes;
+attribute float instanceAngles;
+attribute vec4 instanceColors;
+attribute vec3 instancePickingColors;
+attribute vec4 instanceIconFrames;
+attribute float instanceColorModes;
+attribute vec2 instanceOffsets;
+attribute float instanceLetterIndexInString;
+attribute float instanceStringLength;
+
+uniform vec2 viewportSize;
+uniform float sizeScale;
+uniform float renderPickingBuffer;
+uniform vec2 iconsTextureDim;
+
+varying float vColorMode;
+varying vec4 vColor;
+varying vec2 vTextureCoords;
+
+// rotation should factor in the aspect ratio
+vec2 rotate_by_angle(vec2 vertex, float angle, float aspectRatio) {
+  float angle_radian = angle * PI / 180.0;
+  float cos_angle = cos(angle_radian);
+  float sin_angle = sqrt(1.0 - cos_angle * cos_angle);
```

Comment:
This is actually a bug from my previous version of code. Calculating sin like this will make it always positive... limiting the rotating angle to be always in 0-180. let's just use `sin_angle = sin(angle_radian)`